streams/jvm/src/main/scala/zio/stream/experimental/StringDecoder.scala

```diff
@@ -0,0 +1,229 @@
+package zio.stream.experimental
+
+import java.nio.charset.Charset
+import java.nio.charset.StandardCharsets
+import zio.Chunk
+
+object StringDecoder {
+
+  implicit class ByteStreamDecode[R, E, A](val stream: ZStream[R, E, Byte]) extends AnyVal {
+
+    def utfDecode: ZStream[R, E, String]     = new ZStream[R, E, String](stream.channel >>> StringDecoder.utfDecode)
+    def utf8Decode: ZStream[R, E, String]    = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf8Decode)
+    def utf16Decode: ZStream[R, E, String]   = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf16Decode)
+    def utf16BEDecode: ZStream[R, E, String] = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf16BEDecode)
+    def utf16LEDecode: ZStream[R, E, String] = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf16LEDecode)
+    def utf32Decode: ZStream[R, E, String]   = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf32Decode)
+    def utf32BEDecode: ZStream[R, E, String] = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf32BEDecode)
+    def utf32LEDecode: ZStream[R, E, String] = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf32LEDecode)
+    def usASCIIDecode: ZStream[R, E, String] = new ZStream[R, E, String](stream.channel >>> StringDecoder.usASCIIDecode)
+
+  }
+
+  def utfDecode[Err, Done]: ZChannel[Any, Err, Chunk[Byte], Done, Err, Chunk[String], Done] =
+    ZChannel.readOrIdentity { chunk =>
+      if (chunk.startsWith(0 :: 0 :: -2 :: -1 :: Nil) && Charset.isSupported("UTF-32BE"))
+        utf32BEDecode(chunk.drop(4))
+      else if (chunk.startsWith(-2 :: -1 :: 0 :: 0 :: Nil) && Charset.isSupported("UTF-32LE"))
+        utf32LEDecode(chunk.drop(4))
+      else if (chunk.startsWith(-17 :: -69 :: -65 :: Nil))
+        utf8Decode(chunk.drop(3))
+      else if (chunk.startsWith(-2 :: -1 :: Nil))
+        utf16BEDecode(chunk.drop(2))
+      else if (chunk.startsWith(-1 :: -2 :: Nil))
+        utf16LEDecode(chunk.drop(2))
+      else utf8Decode(chunk)
+    }
+
+  /**
+   * Decodes chunks of UTF-8 bytes into strings.
+   *
+   * This channel uses the String constructor's behavior when handling malformed byte
+   * sequences.
+   */
+  def utf8Decode[Err, Done]: ZChannel[Any, Err, Chunk[Byte], Done, Err, Chunk[String], Done] =
+    utf8Decode(Chunk.empty)
+
+  def utf8Decode[Err, Done](initial: Chunk[Byte]): ZChannel[Any, Err, Chunk[Byte], Done, Err, Chunk[String], Done] = {
```

Comment:
Does this overload need to be public? Could be an inner method on `utf8Decode` perhaps.

Reply:
Also, now that this is here, we can delete `ZSink.utf8Decode`.

Reply:
Right, it should be as private as the other ones.